internal/resources/metal/connection/resource.go

```diff
@@ -0,0 +1,480 @@
+package connection
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"reflect"
+	"strconv"
+
+	equinix_errors "github.com/equinix/terraform-provider-equinix/internal/errors"
+	"github.com/equinix/terraform-provider-equinix/internal/framework"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/packethost/packngo"
+	"golang.org/x/exp/slices"
+)
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: framework.NewBaseResource(
+			framework.BaseResourceConfig{
+				Name:   "equinix_metal_connection",
+				Schema: &resourceSchema,
+			},
+		),
+	}
+}
+
+type Resource struct {
+	framework.BaseResource
+}
+
+func (r *Resource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+
+	var plan ResourceModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Retrieve the API client from the provider metadata
+	r.Meta.AddFwModuleToMetalUserAgent(ctx, req.ProviderMeta)
+	client := r.Meta.Metal
+
+	createRequest, diags := generateCreateRequest(ctx, plan)
+	if diags != nil {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	// Create API resource
+	var conn *packngo.Connection
+	var err error
+	projectId := plan.ProjectID.ValueString()
+
+	// Shared connection specific logic
+	if plan.Type.ValueString() == string(packngo.ConnectionShared) {
+		if projectId == "" {
+			resp.Diagnostics.AddError(
+				"Missing project_id",
+				"project_id is required for 'shared' connection type",
+			)
+			return
+		}
+
+		// Create the shared connection
+		var err error
+		conn, _, err = client.Connections.ProjectCreate(projectId, createRequest)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error creating Metal Connection",
+				"Could not create a shared Metal Connection: "+err.Error(),
+			)
+			return
+		}
+	}
+
+	// Dedicated connection specific logic
+	if plan.Type.ValueString() == string(packngo.ConnectionDedicated) {
+		organizationId := plan.OrganizationID.ValueString()
+		if organizationId == "" {
+			proj, _, err := client.Projects.Get(projectId, &packngo.GetOptions{Includes: []string{"organization"}})
+			if err != nil {
+				resp.Diagnostics.AddError(
+					fmt.Sprintf("Failed to get Project %s", projectId),
+					err.Error(),
+				)
+				return
+			}
+			organizationId = proj.Organization.ID
+		}
+
+		// Create the dedicated connection
+		conn, _, err = client.Connections.OrganizationCreate(organizationId, createRequest)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error creating Metal Connection",
+				"Could not create Metal Connection: "+err.Error(),
+			)
+			return
+		}
+	}
+
+	// Parse API response into the Terraform state
+	resp.Diagnostics.Append(plan.parse(ctx, conn)...)
```

Comment:
The SDK version of this resource called the read function at the end of the create function.  Framework discourages that, but we should probably either (1) replicate the behavior by making a `client.Connections.Get` call to ensure we have all the necessary data or (2) add `include` parameters to the two create requests, assuming the underlying API operations support include parameters.